oauthproxy.go

```diff
@@ -821,6 +821,12 @@ func (p *OAuthProxy) OAuthCallback(rw http.ResponseWriter, req *http.Request) {
 
 // AuthenticateOnly checks whether the user is currently logged in
 func (p *OAuthProxy) AuthenticateOnly(rw http.ResponseWriter, req *http.Request) {
+	isPreflightRequestAllowed := p.skipAuthPreflight && req.Method == "OPTIONS"
+	if isPreflightRequestAllowed {
+		rw.WriteHeader(http.StatusAccepted)
+		return
+	}
```

Comment:
Why not check if it's a whitelisted request, then allow if it is? Would this not have the desired effect and include the IP whitelisting checks as well?